doc/shelley-genesis.md

```diff
@@ -392,127 +495,827 @@ It follows this file layout convention:
+                           Output filepath of the signing key.
+```
+
+So let's do that too
+```
+$ cardano-cli shelley node key-gen-VRF \
+    --verification-key-file example/node1/vrf.vkey \
+    --signing-key-file example/node1/vrf.skey
+
+$ cardano-cli shelley node key-gen-VRF \
+    --verification-key-file example/node2/vrf.vkey \
+    --signing-key-file example/node2/vrf.skey
+```
+
+### Issuing an operational certificate
 
-## Creating node operational keys
+Now we get to the stage of wanting to issue an operational certificate.
 
-TODO
+When doing this for real, the operator's offline key should of course be
+offline, so we would issue the certificate on the offline machine with
+the offline key, and copy the resulting certificate to the operational machine.
 
-## Issuing node operational certificates
+```
+$ cardano-cli shelley node issue-op-cert
+Usage: cardano-cli shelley node issue-op-cert --hot-kes-verification-key-file FILEPATH
+                                              --cold-signing-key-file FILEPATH
+                                              --operational-certificate-issue-counter FILE
+                                              --kes-period NATURAL
+                                              --out-file FILE
+  Issue a node operational certificate
 
-TODO
+Available options:
+  --hot-kes-verification-key-file FILEPATH
+                           Filepath of the hot KES verification key.
+  --cold-signing-key-file FILEPATH
+                           Filepath of the cold signing key.
+  --operational-certificate-issue-counter FILE
+                           The file with the issue counter for the operational
+                           certificate.
+  --kes-period NATURAL     The start of the KES key validity period.
+  --out-file FILE          The output file.
+
+```
+There's a few things here to understand.
+
+As discussed above, a certificate identifies an operational KES key that we
+will be using to sign block headers, so we need is verification key. It is
+signed by the operator's offline key so we need that signing key.
+
+As mentioned, certificates have an issue counter number that is used to
+inform other nodes that older certificates are now invalid. This certificate
+issue counter must be kept with the operator's offline key. You'll notice they
+got created when we created the genesis delegate keys, or if you create a new
+stake pool key.
+
+Finally we have a confusing flag for the KES period. Each operational
+certificate specifies when the certificate is valid from. This is like a date
+but is specified in terms of KES periods, which is some number of slots long.
+Frankly, this needs improving in the CLI tools and/or documentation. For now
+we are creating a system from scratch so we can start with period 0.
+
+So let's go ahead and issue ourselves an operational certificate. We will sign
+using use the genesis delegate keys we created earlier.
+```
+$ cardano-cli shelley node issue-op-cert \
```

Comment:
It seem that the issue is that I don't have any `delegate-opcert1.counter` in example/delegate-keys/ . Maybe I should use `delegate1.counter` instead of `delegate-opcert1.counter` ?
```
dorin@dorin-Virtual-Machine:~/cardano-node/cardano-cli$ cabal v2-run -- cardano-cli shelley node issue-op-cert --hot-kes-verification-key-file example/node1/kes.vkey --cold-signing-key-file example/delegate-keys/delegate1.skey --operational-certificate-issue-counter example/delegate-keys/delegate-opcert1.counter --kes-period 0 --out-file example/node1/cert
Up to date
ReadOperationalCertIssueCounterError (TextViewFileIOError "example/delegate-keys/delegate-opcert1.counter" example/delegate-keys/delegate-opcert1.counter: openBinaryFile: does not exist (No such file or directory))
```
```
$ ls example/delegate-keys/
delegate1.counter  delegate1.skey  delegate1.vkey
```

Reply:
oops, thanks